### PR TITLE
cor: Fix read out-of-band data

### PR DESCRIPTION
**Pull request by devimc**
Sometimes when cc-proxy sends only out-of-band
data cor tries to convert it to big endian
returning a negative value that is used to
append the iov buffer to msg_received
causing a crash (segment default)
    
To fix that crash this patch validates that
messages read from cc-proxy are not out-of-band data.

### CODE REVIEW

**Comment by chavafg** [comment]
qa-failed

File: src/proxy.c [hunk 1]
```diff
@@ -277,12 +273,26 @@ cc_proxy_msg_read (GIOChannel *source, GIOCondition condition,
...
-	bytes_to_copy = MIN((size_t)bytes_read-data_offset, (size_t)p_msg.length);
-	g_string_append_len(proxy_data->msg_received,
-		iov_buffer+data_offset, (ssize_t)bytes_to_copy);
-
-	if (bytes_read >= p_msg.length+data_offset) {
+	/* cc-proxy sends a fd as out-of-band data
+	 * followed by an 'F'.
+	 * *DO NOT* copy these characters to msg_received
+	 */
+	if (bytes_read > 1 || iov_buffer[0] != 'F') {
```
Comment (jcvenegas) [inline src/proxy.c]: so, when bytes_read is > 1  iov_buffer[0] will never be 'F'  (well will be too big ), right? 
lgtm

File: src/proxy.c [hunk 2]
```diff
@@ -277,12 +273,26 @@ cc_proxy_msg_read (GIOChannel *source, GIOCondition condition,
...
+		g_string_append_len (proxy_data->msg_received,
+			iov_buffer+data_offset, (ssize_t)bytes_to_copy);
+
+		/* all message was read */
+		if (bytes_read >= p_msg.length+data_offset) {
+			goto out;
+		}
+	} else {
+		/* only out-of-band data was received */
 		goto out;
```
Comment (jodh-intel) [inline src/proxy.c]: Is this `goto` safe? It jumps down to a `g_debug()` call that displays the received message... but there isn't one in the OOB message scenario right? Do we need a new `out2` label maybe?

**Comment by sboeuf** [comment]
lgtm

**Comment by amshinde** [comment]
lgtm

**Pull request merged**